manager/controlapi/node.go

```diff
@@ -273,7 +273,8 @@ func (s *Server) RemoveNode(ctx context.Context, request *api.RemoveNodeRequest)
 		if node == nil {
 			return grpc.Errorf(codes.NotFound, "node %s not found", request.NodeID)
 		}
-		if node.Spec.Role == api.NodeRoleManager {
+		member := s.raft.GetMemberByNodeID(request.NodeID)
```

Comment:
It shouldn't be `nil` in a real environment. But the check at the beginning of the function suggests that it is called by a unit test that doesn't set up a raft cluster, or was at one time. It's probably a good idea to have the same check here.